test/langtools/tools/javac/8278078/EnclosingSuperTest.java

```diff
@@ -0,0 +1,13 @@
+/**
+ * @test /nodynamiccopyright/
+ * @bug 8278078
+ * @summary error: cannot reference super before supertype constructor has been called
+ * @run compile EclosingSuperTest.java
+ */
+public class EnclosingSuperTest {
```

Comment:
My suggestion for another test is something like this:

```
public class EnclosingSuperTest2 {
    class InnerClass extends Exception {
        InnerClass() {
            this(EnclosingSuperTest2.super.toString());
        }
        
        InnerClass(String s) { }
    }
}
```

This test also pass before the problematic fix, and fails afterwards, so I think we should add it too.

Reply:
Yes, now I understand.
I've extended set of "Valid" test cases and also added several "Invalid" test cases expecting to fail to compile.
During that work I found <self class>.super used in constructor arg is not failing compilation, so the Attr patch is also adjusted for that case.

However all the test cases here are still related to this and super in constructor args only, test matrix for other contexts should follow.